src/transforms/sample/config.rs

```diff
@@ -29,7 +29,16 @@ pub struct SampleConfig {
     /// For example, `rate = 1500` means 1 out of every 1500 events are forwarded and the rest are
     /// dropped.
     #[configurable(metadata(docs::examples = 1500))]
-    pub rate: u64,
+    #[configurable(deprecated = "This option will be deprecated, prefer percent_rate instead")]
+    pub rate: Option<u64>,
+
+    /// The rate at which events are forwarded, expressed as a percentage
+    ///
+    /// For example, `ratio = .13` means that 13% out of all events on the stream are forwarded and
+    /// the rest are dropped.
+    #[configurable(metadata(docs::examples = "0.13"))]
+    #[configurable(validation(range(min = 0.0, max = 1.0)))]
+    pub percent_rate: f32,
```

Comment:
I had another idea for the configuration here, which admittedly might be too confusing to live: Keep this as just a `rate` setting, but make it a float. When it is greater than 1, treat it as a 1/N fraction, and treat it as a ratio otherwise. Still just one setting but adds different interpretations. Exactly 1 works the same in both interpretations. Again, this might lead to more confusion rather than less, so feel free to reject it in favor of a pair of `Option`s.

Reply:
>  I had another idea for the configuration here, which admittedly might be too confusing to live

😄 I'm very strongly against keeping this as a single `rate` setting.  We want less confusion for users on this 👍 We've had several asks from customers for a percentage and I would present it that way.

Reply:
Yes I would also be in favor of another variable, in my opinion a configuration variable that behaves differently depending on its numeric value would be confusing